packages/stellar-axelar-std/src/interfaces/upgradable.rs

```diff
@@ -16,16 +16,22 @@ pub trait UpgradableInterface: OwnableInterface {
     fn upgrade(env: &Env, new_wasm_hash: BytesN<32>);
 }
 
-pub trait MigratableInterface: UpgradableInterface {
-    /// Data needed during the migration. Each contract can define its own data type.
-    type MigrationData: FromVal<Env, Val>;
+pub trait MigratableInterface: UpgradableInterface + CustomMigratableInterface {
     /// Error type returned if the migration fails.
     type Error: Into<soroban_sdk::Error>;
 
     /// Migrates contract state after upgrading the contract code.
     fn migrate(env: &Env, migration_data: Self::MigrationData) -> Result<(), Self::Error>;
 }
 
+pub trait CustomMigratableInterface: UpgradableInterface {
+    /// Data needed during the migration. Each contract can define its own data type.
+    type MigrationData: FromVal<Env, Val>;
+
+    /// Migrates contract state after upgrading the contract code.
+    fn __migrate(env: &Env, migration_data: Self::MigrationData);
```

Comment:
```suggestion
    /// Data needed during the migration. Each contract can define its own data type.
    type MigrationData: FromVal<Env, Val> = ();

    /// Migrates contract state after upgrading the contract code.
    fn __migrate(env: &Env, migration_data: Self::MigrationData) {}
```
Let's have a default implementation that can be overridden when needed?

Reply:
yeah good idea